### PR TITLE
Fix mass ws deletion bug

### LINKED ISSUE #17729: Fix Mantid crash when closing many workspaces via MantidDock
Rob mentioned that closing about 200 workspaces at once via MantidDock causes a crash. 

**Solution**
Avoid deleting the workspaces via the `DeleteWorkspace` algorithm and clear them instead directly from the ADS.

### PR DESCRIPTION
**Pull request by NickDraper**
Fixes #17729

Added a DeleteWorkspaces algorithm, use it in GUI when deleting multiple workspaces, all it does is iterate over the list deleting workspaces.

**To test:**

*\* Test that deleting many workspaces does not crash**
1. Create many workspaces with:

``` python
for i in range(3000):
    name = "WS_" + str(i)
    CreateSampleWorkspace(OutputWorkspace=name)
```
1. Select all in the ADS
2. Press the delete button
   - Confirm that Mantid does not crash and that the time it takes to clear the 3000 workspaces is  acceptable.

*\* Test that deleting few very large data sets does not disturb the user experience (too much)**
1. Find a sample data set here: smb://olympic/babylon5/Scratch/Anton/SampleWS . The relevant file is MDEvent_ellipPeaks_new.nxs
1. If you have a 16GB computer run the follow script, else you have to modify the times you can load the file.

``` python
for i in range(10):
    name = "WS_" + str(i)
    Load(Filename='PATH_TO/MDEvent_ellipPeaks_new.nxs', OutputWorkspace=name)
```
1. Select all workspace and delete them.
   - Confirm that the time taken is acceptable and that the screen does not freeze. It will be unusable though.

### RELEASE NOTES

docs/source/release/framework.rst

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidPlot/src/Mantid/MantidUI.cpp [hunk 1]
```diff
@@ -455,9 +455,15 @@ void MantidUI::deleteWorkspaces(const QStringList &wsNames) {
...
-        deleteWorkspace(ws);
+      auto alg = createAlgorithm("DeleteWorkspaces");
+      alg->setLogging(false);
+      std::vector<std::string> vecWsNames;
+      vecWsNames.reserve(wsNames.size());
+      foreach (auto wsName, wsNames) {
+        vecWsNames.push_back(wsName.toStdString());
       }
+      alg->setProperty("WorkspaceList", vecWsNames);
+      executeAlgorithmAsync(alg);
```
Comment (samueljackson92) [inline MantidPlot/src/Mantid/MantidUI.cpp]: Maybe we should put all this in a private method? `void MantidUI::deleteWorkspaces()` or something?

**Review by samueljackson92 — CHANGES_REQUESTED** [review]
Tested with example scripts and played with tying to pass invalid input (e.g. no workspaces) to algorithm. All seems well. See comment below from code review but this is basically ready to go I think.

**Comment by samueljackson92** [comment]
@NickDraper Also checked this with Giovanni's Mantid project and I cannot reproduce.

File: MantidPlot/src/Mantid/MantidUI.cpp (hunk 1 above)
Comment (NickDraper) [inline MantidPlot/src/Mantid/MantidUI.cpp]: it already is!

Comment (samueljackson92) [inline MantidPlot/src/Mantid/MantidUI.cpp]: So it is!

**Review by samueljackson92 — APPROVED** [review]

**Comment by samueljackson92** [comment]
LGTM :shipit:

**Pull request merged**